src/viewProviders/schemas.test.ts

```diff
@@ -120,6 +122,67 @@ describe("SchemasViewProvider setSchemaRegistry()", () => {
       assert.ok(setSchemaRegistryFake.calledWith(newRegistry));
     }
   });
+
+  it("Firing environmentChanged + deleted should call reset()", async () => {
+    const resetFake = sandbox.fake();
+    sandbox.replace(provider, "reset", resetFake);
+
+    // Be set to a SR within the environment being deleted
+    provider.schemaRegistry = TEST_LOCAL_SCHEMA_REGISTRY;
+    // fire the event
+    environmentChanged.fire({ id: TEST_LOCAL_ENVIRONMENT_ID, wasDeleted: true });
+
+    // Should have called .reset()
+    assert.ok(resetFake.calledOnce);
+  });
+
+  it("Firing environmentChanged + misc change should not call reset(), should call updateTreeViewDescription + refresh", async () => {
+    const resetFake = sandbox.fake();
+    const updateTreeViewDescriptionFake = sandbox.fake();
+    const refreshFake = sandbox.fake();
+
+    sandbox.replace(provider, "reset", resetFake);
+    sandbox.replace(provider, "updateTreeViewDescription", updateTreeViewDescriptionFake);
+    sandbox.replace(provider, "refresh", refreshFake);
+
+    // Be set to a SR within the environment being deleted
+    provider.schemaRegistry = TEST_LOCAL_SCHEMA_REGISTRY;
+    // fire the event
+    environmentChanged.fire({ id: TEST_LOCAL_ENVIRONMENT_ID, wasDeleted: false });
+
+    // Need to pause an iota to get the refresh to be called, is after first await in the block.
+    await pause(100);
```

Comment:
Could we use https://sinonjs.org/releases/latest/fake-timers/ here? We have a couple other test suites with it:
https://github.com/confluentinc/vscode/blob/8b63d371250cfb801adecec0566f57c4e71d41c2/src/sidecar/connections/watcher.test.ts#L83